python/lsst/ts/aos/utils/hexapods/hexapod_lut.py

```diff
@@ -0,0 +1,217 @@
+import argparse
+import asyncio
+
+import numpy as np
+import pandas as pd
+import yaml
+from astropy.time import Time
+from lsst_efd_client import EfdClient
+from numpy.polynomial import Polynomial
+
+
+def extract_component_data(offsets: pd.DataFrame, component: str) -> pd.DataFrame:
+    """Extracts the component data from the offsets dataframe
+
+    Parameters
+    ----------
+    offsets: pandas.DataFrame
+        Dataframe with the offsets data from the EFD
+    component: str
+        Component to extract the data for. Can be 'M2', 'Camera'
+
+    Returns
+    -------
+    pandas.DataFrame
+        Dataframe with the offsets data for the specified component
+    """
+
+    def extract_target_angles(x: str) -> list[float]:
+        """Extracts the target angles from the target string"""
+        parts = x.split("_")
+        return [float(parts[1]), float(parts[2]), float(parts[3])]
+
+    if component == "Camera":
+        offsets = offsets.loc[offsets["target"].str.startswith("FrameCAM")].copy()
+    elif component == "M2":
+        offsets = offsets.loc[offsets["target"].str.startswith("FrameM2")].copy()
+
+    target_angles = offsets["target"].apply(lambda x: extract_target_angles(x))
```

Comment:
lambdas are not allowed per the DM dev guidelines: https://developer.lsst.io/python/style.html#lambda-should-not-be-used